contracts/axelar-auth-verifier/src/test.rs

```diff
@@ -79,6 +251,155 @@ fn test_transfer_operatorship() {
+
+    let user = Address::generate(&env);
+    let previous_signer_retention = 1;
+
+    initialize(
+        &env,
+        &client,
+        user.clone(),
+        previous_signer_retention,
+        randint(1, 10),
+    );
+
+    let new_signers = generate_signer_set(&env, randint(1, 10), true);
+
+    let encoded_new_signer_set = new_signers.signer_set.to_xdr(&env);
+
+    // should throw an error, last signer weight is zero
+    let res = client.try_transfer_operatorship(&encoded_new_signer_set);
+    assert!(res.is_err());
+}
+
+#[test]
+fn test_transfer_operatorship_fail_zero_threshold() {
+    let (env, _, client) = setup_env();
+
+    let user = Address::generate(&env);
+    let previous_signer_retention = 1;
+
+    initialize(
+        &env,
+        &client,
+        user.clone(),
+        previous_signer_retention,
+        randint(1, 10),
+    );
+
+    let mut new_signers = generate_signer_set(&env, randint(1, 10), false);
+
+    // set the threshold to zero
+    new_signers.signer_set.threshold = U256::from_u32(&env, 0);
+
+    let encoded_new_signer_set = new_signers.signer_set.to_xdr(&env);
+
+    // should error because the threshold is set to zero
+    let res = client.try_transfer_operatorship(&encoded_new_signer_set);
+    assert!(res.is_err());
+}
+
+#[test]
+fn test_transfer_operatorship_fail_low_total_weight() {
+    let (env, _, client) = setup_env();
+
+    let user = Address::generate(&env);
+    let previous_signer_retention = 1;
+
+    initialize(
+        &env,
+        &client,
+        user.clone(),
+        previous_signer_retention,
+        randint(1, 10),
+    );
+
+    let mut new_signers = generate_signer_set(&env, randint(1, 10), false);
+
+    let env = &env;
+    let zero = U256::from_u32(env, 0);
+    let one = U256::from_u32(env, 1);
+    let mut total_weight = zero.clone();
+
+    for weight in new_signers.signer_set.signers.iter().map(|s| s.1) {
+        total_weight = total_weight.add(&weight);
+    }
+
+    let new_threshold = total_weight.add(&one);
+
+    // set the threshold to zero
+    new_signers.signer_set.threshold = new_threshold;
+
+    let encoded_new_signer_set = new_signers.signer_set.to_xdr(&env);
+
+    // should error because the threshold is set to zero
+    let res = client.try_transfer_operatorship(&encoded_new_signer_set);
+    assert!(res.is_err());
+}
+
+#[test]
+fn test_transfer_operatorship_fail_wrong_signer_order() {
+    let (env, _, client) = setup_env();
+
+    let user = Address::generate(&env);
+    let previous_signer_retention = 1;
+
+    initialize(
+        &env,
+        &client,
+        user.clone(),
+        previous_signer_retention,
+        randint(1, 10),
+    );
+
+    let mut new_signers = generate_signer_set(&env, randint(1, 10), false);
+
+    let len = new_signers.signer_set.signers.len();
+
+    // create a new vec and reverse signer order
+    let mut reversed_signers = Vec::new(&env);
+    for i in (0..len).rev() {
```

Comment:
use functional approach. i recommend keeping the std library doc at handy. lot of useful functions available there, such as reversing a vec

Reply:
Makes sense but unless I'm wrong the Soroban sdk vec implementation doesn't have a lot of the functions found in the std library such as reverse, collect, etc:
https://docs.rs/soroban-sdk/latest/soroban_sdk/struct.Vec.html

Reply:
You can convert to/from std Vec. Add the alloc feature to the dependency and you can use some more methods